nemo/collections/asr/data/text_to_text.py

```diff
@@ -0,0 +1,462 @@
+        audio_signal = pad_sequence([item[0] for item in asr_items], batch_first=True, padding_value=0.0)
+        a_sig_length = torch.tensor([item[1] for item in asr_items]).long()
+
+        tts_texts = pad_sequence(
+            [item.tts_text for item in text_items], batch_first=True, padding_value=tts_text_pad_id
+        )
+        tts_text_length = torch.tensor([item.tts_text.shape[0] for item in text_items]).long()
+        speakers = torch.tensor([item.speaker for item in text_items]).long()
+
+        transcripts = pad_sequence(
+            [item.transcript for item in text_items] + [item[2] for item in asr_items],
+            batch_first=True,
+            padding_value=asr_pad_id,
+        )
+        transcript_length = torch.tensor(
+            [item.transcript.shape[0] for item in text_items] + [item[3] for item in asr_items]
+        ).long()
+
+        return TextOrAudioToTextBatch(
+            audio_signal=audio_signal,
+            a_sig_length=a_sig_length,
+            tts_texts=tts_texts,
+            tts_text_length=tts_text_length,
+            speakers=speakers,
+            transcripts=transcripts,
+            transcript_length=transcript_length,
+        )
+
+
+def _asr_text_to_tokens(text: str) -> np.ndarray:
+    """
+    Helper function for asr tokenization with multiprocessing pool only.
+    Must be defined on the top level.
+    Expects asr_tokenizer_global, asr_bos_id_global, asr_eos_id_global to exist in the current pool process
+    """
+    ids = asr_tokenizer_global.text_to_ids(text)
+    if asr_bos_id_global is not None:
+        ids = [asr_bos_id_global] + ids
+    if asr_eos_id_global is not None:
+        ids.append(asr_eos_id_global)
+    return np.asarray(ids)
+
+
+def _tts_text_to_tokens(text: str) -> np.ndarray:
+    """
+    Helper function for asr tokenization with multiprocessing pool only.
+    Must be defined on the top level.
+    Expects tts_tokenizer_global to exist in the current pool process
+    """
+    return np.asarray(tts_tokenizer_global(text))
+
+
+def _iterate_manifest(filepath: AnyPath) -> Iterable[Dict[str, Any]]:
+    """
+    Helper function to iterate manifest
+    """
+    with open(filepath, "r", encoding="utf-8") as f:
+        for line in f:
+            record = json.loads(line)
+            yield record
+
+
+class TextToTextDatasetBase:
+    """
+    Base class for loading text-to-text manifests
+    Map-style and Iterable datasets should inherit this class
+    """
+
+    asr_pad_id: int
+    tts_text_pad_id: int
+    asr_bos_id: Optional[int] = None
+    asr_eos_id: Optional[int] = None
+
+    def __init__(
+        self,
+        manifest_filepath: Union[AnyPath, List[AnyPath]],
+        speakers_filepath: Union[AnyPath, List[AnyPath]],
+        asr_tokenizer: TokenizerSpec,
+        asr_use_start_end_token: bool,
+        tts_parser: Callable,
+        tts_text_pad_id: int,
+        tts_text_normalizer: Normalizer,
+        tts_text_normalizer_call_kwargs: Dict,
```

Comment:
I wonder if there's a better way to pass tts arguments rather than via fake kwargs

Reply:
These kwargs are taken from TTS model directly, and since this dataset is only used in ASRWithTTSModel, I don't like an idea of unpacking them here.